### PR TITLE
Improve json sql functions by serializing json element into column's buffer direclty

### PR DESCRIPTION
**Pull request by lgbo-ustc**
### Changelog category (leave one):

- Performance Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):

Instead of serializing json elements into a `std::stringstream`, this PR try to put the serialization result into `ColumnString` direclty.

test result for
```SQL
insert into t_json_str select concat('{"x":', cast(number as String), ',"y":"helloeee", "z":345, "a":{"k1":1, "k2":2}}') from numbers(10000000);
select JSON_QUERY(x, '$.x') from t_json_str Format Null;
```

| | result |
| --- | --- |
| before | 0 rows in set. Elapsed: 3.981 sec. Processed 10.00 million rows, 678.89 MB (2.51 million rows/s., 170.51 MB/s.)|
| after | 0 rows in set. Elapsed: 1.570 sec. Processed 10.00 million rows, 678.89 MB (6.37 million rows/s., 432.43 MB/s.)|

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit cd8a8980f821dcecb1e15bf40043220c4288c0a4 with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/54613/cd8a8980f821dcecb1e15bf40043220c4288c0a4/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Mergeable Check</td><td>Checks if all other necessary checks are successful</td><td>✅ success</td></tr>
<tbody>
</table>
</details>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td>⏳ pending</td></tr>
<tbody>
</table>

File: src/Common/JSONParsers/SimdJSONParser.h [hunk 1]
```diff
@@ -16,6 +18,251 @@ namespace ErrorCodes
     extern const int CANNOT_ALLOCATE_MEMORY;
 }
 
+class SimdJSONStringFormatter
```
Comment (antaljanosbenjamin) [inline src/Common/JSONParsers/SimdJSONParser.h:21]: Please add a big comment to these classes about their origin implementation is in simdjson. I found out this accidentally, but actually it is really important detail.

Furthermore, please adjust the casing of the functions to follow `camelCasing` as the rest of our code base.

**Review by antaljanosbenjamin — CHANGES_REQUESTED** [review]

File: tests/performance/json_extract_simdjson.xml [hunk 2]
```diff
@@ -33,4 +33,8 @@
     <query>SELECT 'simdjson-12', count() FROM zeros(5000000) WHERE NOT ignore(JSONExtractFloat(materialize({long_json}), 'fparam', 'nested_2', -2))</query>
     <query>SELECT 'simdjson-13', count() FROM zeros(5000000) WHERE NOT ignore(JSONExtractBool(materialize({long_json}), 'bparam'))</query>
 
+    <query>SELECT 'simdjson-13', count() FROM zeros(5000000) WHERE NOT ignore(JSON_VALUE(materialize({long_json}), '$.nparam'))</query>
+    <query>SELECT 'simdjson-13', count() FROM zeros(5000000) WHERE NOT ignore(JSON_QUERY(materialize({long_json}), '$.nparam'))</query>
+    <query>SELECT 'simdjson-13', count() FROM zeros(5000000) WHERE NOT ignore(JSON_VALUE(materialize({json}), '$.nparam'))</query>
+    <query>SELECT 'simdjson-13', count() FROM zeros(5000000) WHERE NOT ignore(JSON_QUERY(materialize({json}), '$.nparam'))</query>
```
Comment (antaljanosbenjamin) [inline tests/performance/json_extract_simdjson.xml:39]:
```suggestion
    <query>SELECT 'simdjson-14', count() FROM zeros(5000000) WHERE NOT ignore(JSON_VALUE(materialize({long_json}), '$.nparam'))</query>
    <query>SELECT 'simdjson-15', count() FROM zeros(5000000) WHERE NOT ignore(JSON_QUERY(materialize({long_json}), '$.nparam'))</query>
    <query>SELECT 'simdjson-16', count() FROM zeros(5000000) WHERE NOT ignore(JSON_VALUE(materialize({json}), '$.nparam'))</query>
    <query>SELECT 'simdjson-17', count() FROM zeros(5000000) WHERE NOT ignore(JSON_QUERY(materialize({json}), '$.nparam'))</query>
```

**Review by antaljanosbenjamin — COMMENTED** [review]
Wow, I really like the new version with the serializers, pretty awesome, thanks!

Just a minor comment, after that it should be mergeable.

**Review by antaljanosbenjamin — APPROVED** [review]

**Pull request merged**